src/aal/aal.h

```diff
@@ -23,6 +23,15 @@
 #  define PLATFORM_IS_POWERPC
 #endif
 
+#if defined(__mips__)
+#  define PLATFORM_IS_MIPS
+#endif
+
+#if defined(__riscv)
+#  if __riscv_xlen == 64
+#    define PLATFORM_IS_RV64
+#  endif
```

Comment:
I don't see why this needs to care about XLEN. The generic AAL bits don't care about 32-bit vs 64-bit X86/Arm/PowerPC/MIPS (though some of those probably lack 32-bit support). Just have a PLATFORM_IS_RISCV and an `aal_riscv.h`, with SNMALLOC_VA_BITS_32 vs SNMALLOC_VA_BITS_64 conditioned on `__riscv_xlen` inside the header itself?